Go/ds/singly_linked_list_reversed.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Node struct {
+	value int
+	next  *Node
+}
+type List struct {
+	head *Node
+	tail *Node
+	size int
+}
+
+//makeLinkedList: creates an empty linked list
+func makeLinkedList() List {
+	var node *Node = nil
+	l := List{node, node, 1}
+	return l
+}
+
+//createAtFront: creates a node at the frount of the list
+func (list *List) createAtFront(data int) {
+	var node Node
+	node.value = data
+	node.next = list.head
+	if list.tail == nil {
+		list.tail = &node
+	}
+	list.head = &node
+	fmt.Printf("created a node with the value %d at the frount of the list\n", data)
+}
+
+//printList: prints out the linked list from head to tail
+func (list *List) printList() {
+	currentNode := list.head
+	for currentNode != nil {
+		fmt.Printf("%d->", currentNode.value)
+		currentNode = currentNode.next
+	}
+	fmt.Printf("\n")
+}
+
+//reverseLinkedList: reverses the linked list
+func (list *List) reverseLinkedList() {
+	currentNode := list.head
+	var prev *Node = nil
+	next := currentNode.next
+	for currentNode != nil {
+		next = currentNode.next
+		currentNode.next = prev
+		prev = currentNode
+		currentNode = next
+	}
+	list.tail = list.head
+	list.head = prev
+	fmt.Println("successfully reversed the linked list")
+}
+
+//fillListWithDummyData: add values 1 - 9 to the linked list for testing reasons
```

Comment:
Please use better comments to exactly show what is being done, rather than vague idea about the same